src/dataset.js

```diff
@@ -69,13 +100,21 @@ export default class Dataset {
     }
   }
 
-  setReadOffset(readOffset) {
-    if (this.state.readOffset === readOffset) { return; }
+
+  setReadOffset(readOffset, options = {}) {
```

Comment:
The `setReadOffset` function seems too overloaded here. Is there any reason to not have `setReadOffset`, `refresh`, and `refilter` just do only those things?